openff/toolkit/topology/molecule.py

```diff
@@ -2040,6 +2041,20 @@ def __hash__(self):
         """
         return hash(self.to_smiles())
 
+    #@cached_property
+    def ordered_connection_table_hash(self):
+        if self._ordered_connection_table_hash is not None:
+            return self._ordered_connection_table_hash
+        
+        id = ''
+        for atom in self.atoms:
+            id += f'{atom.element.symbol}_{atom.formal_charge}_{atom.stereochemistry}__'
+        for bond in self.bonds:
+            id += f'{bond.bond_order}_{bond.stereochemistry}_{bond.atom1_index}_{bond.atom2_index}__'
+        #return hash(id)
+        self._ordered_connection_table_hash = hash(id)
+        return self._ordered_connection_table_hash
```

Comment:
I need to think more about if this is totally suitable for caching outputs of `to_openeye` and `to_rdkit`. It captures the important attributes of the _connection table_ (so it's safe for SMARTS matching), but it will (incorrectly) return a false positive for molecules with the same connection table but different conformers, partial charges, or names. 

I could potentially either
* Separate the connection table conversion lines in `to_openeye` (ie the atom/bond addition) into a private method, and only apply the caching to that. 
* Expand the cache key to include more information (like molecule name and partial charges), but at that point this will be about as slow as `to_dict`
* Avoid caching altogether by making a multi-SMARTS matching method (instead of running find_smarts_matches repeatedly on one SMARTS each time)